java-connectors/kafka-connect-azure-eventhubs/src/test/java/io/lenses/streamreactor/connect/azure/eventhubs/source/TopicPartitionOffsetProviderTest.java

```diff
@@ -0,0 +1,109 @@
+import io.lenses.streamreactor.connect.azure.eventhubs.source.TopicPartitionOffsetProvider.AzureTopicPartitionKey;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import org.apache.kafka.connect.storage.OffsetStorageReader;
+import org.junit.jupiter.api.Test;
+
+class TopicPartitionOffsetProviderTest {
+
+  @Test
+    void getOffsetShouldCallOffsetStorageReader() {
+      //given
+      OffsetStorageReader offsetStorageReader = mock(OffsetStorageReader.class);
+      TopicPartitionOffsetProvider topicPartitionOffsetProvider = new TopicPartitionOffsetProvider(
+          offsetStorageReader);
+      String topic = "some_topic";
+      Integer partition = 1;
+
+      //when
+      AzureTopicPartitionKey azureTopicPartitionKey = new AzureTopicPartitionKey(topic, partition);
+      topicPartitionOffsetProvider.getOffset(azureTopicPartitionKey);
+
+      //then
+      verify(offsetStorageReader).offset(azureTopicPartitionKey);
+    }
+
+    @Test
+    void getOffsetShouldReturnEmptyOptionalIfCommitsNotFound() {
+      //given
+      OffsetStorageReader offsetStorageReader = mock(OffsetStorageReader.class);
+      when(offsetStorageReader.offset(any(Map.class))).thenReturn(new HashMap());
+      String topic = "some_topic";
+      Integer partition = 1;
+      TopicPartitionOffsetProvider topicPartitionOffsetProvider = new TopicPartitionOffsetProvider(
+          offsetStorageReader);
+
+      //when
+      AzureTopicPartitionKey azureTopicPartitionKey = new AzureTopicPartitionKey(topic, partition);
+      Optional<AzureOffsetMarker> offset = topicPartitionOffsetProvider.getOffset(azureTopicPartitionKey);
+
+      //then
+      verify(offsetStorageReader).offset(azureTopicPartitionKey);
+      assertTrue(offset.isEmpty());
+    }
+
+    @Test
+    void getOffsetShouldReturnValidAzureOffsetMarkerIfCommitsFound() {
+      //given
+      long offsetOne = 1L;
+      String OFFSET_KEY = "OFFSET";
+      OffsetStorageReader offsetStorageReader = mock(OffsetStorageReader.class);
+      HashMap<String, Long> offsets = new HashMap<>();
+      offsets.put(OFFSET_KEY, offsetOne);
+      when(offsetStorageReader.offset(any(Map.class))).thenReturn(offsets);
+      String topic = "some_topic";
+      Integer partition = 1;
+      TopicPartitionOffsetProvider topicPartitionOffsetProvider = new TopicPartitionOffsetProvider(
+          offsetStorageReader);
+
+      //when
+      AzureTopicPartitionKey azureTopicPartitionKey = new AzureTopicPartitionKey(topic, partition);
+      Optional<AzureOffsetMarker> offset = topicPartitionOffsetProvider.getOffset(azureTopicPartitionKey);
+
+      //then
+      verify(offsetStorageReader).offset(azureTopicPartitionKey);
+      assertTrue(offset.isPresent());
+      assertEquals(offsetOne, offset.get().getOffsetValue());
+    }
+
+    @Test
+    void AzureTopicPartitionKeyShouldReturnTopicAndPartitionValues() {
```

Comment:
this should match format of the other test names

Reply:
changed